libpldmresponder/fru.cpp

```diff
@@ -453,6 +436,43 @@ void FruImpl::removeIndividualFRU(const std::string& fruObjPath)
     } // sm00 this can be RECORDS_DELETED also for adapter pdrs
 }
 
+void FruImpl::deleteFruRecord(uint16_t rsi)
+{
+    std::vector<uint8_t> updatedFruTbl;
+    const struct pldm_fru_record_data_format* recordSetSrc =
+        reinterpret_cast<const struct pldm_fru_record_data_format*>(
+            table.data());
+
+    const struct pldm_fru_record_tlv* tlv;
+    size_t pos = 0;
+
+    while (table.size() > pos)
+    {
+        size_t recordLen = sizeof(struct pldm_fru_record_data_format) -
+                           sizeof(struct pldm_fru_record_tlv);
+        tlv = recordSetSrc->tlvs;
+
+        for (uint8_t i = 0; i < recordSetSrc->num_fru_fields; i++)
+        {
+            size_t len = sizeof(*tlv) - 1 + tlv->length;
+            recordLen += len;
+            tlv = reinterpret_cast<const struct pldm_fru_record_tlv*>(
+                (char*)tlv + len);
+        }
+        if ((recordSetSrc->record_set_id != htole16(rsi) && rsi != 0))
+        {
+            std::copy(table.begin() + pos, table.begin() + pos + recordLen,
+                      std::back_inserter(updatedFruTbl));
+        }
```

Comment:
we are copying everything from begin to the postiion , but what about the records that are after the position ?
if the fru to be removed is at 5th postion , we just have to skip 5 an copy the entire table, but looking at this i feel we are copying only till 5th location.

Am i missing something here ?

Reply:
We r running this if block in the while loop till we reach end of the frutable. So it will copy all other FRU records except the rsi matching ones.